baremetal/metal3machine_manager_test.go

```diff
@@ -563,6 +600,168 @@ var _ = Describe("Metal3Machine manager", func() {
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaEmptyAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: true,
+			ExpectError:   false,
+		}),
+	)
+
+	type testCaseCheckPauseAnnotation struct {
+		Cluster       *capi.Cluster
+		BMMachine     *capm3.Metal3Machine
+		Host          *bmh.BareMetalHost
+		ExpectPresent bool
+		ExpectError   bool
+	}
+
+	DescribeTable("Test Remove BMH Pause Annotation",
+		func(tc testCaseCheckPauseAnnotation) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.Host, tc.BMMachine, tc.Cluster)
+			machineMgr, err := NewMachineManager(c, tc.Cluster, nil, nil, tc.BMMachine, klogr.New())
+			Expect(err).NotTo(HaveOccurred())
+
+			err = machineMgr.CheckPauseAnnotation(context.TODO())
+			if tc.ExpectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			savedHost := bmh.BareMetalHost{}
+			err = c.Get(context.TODO(),
+				client.ObjectKey{
+					Name:      tc.Host.Name,
+					Namespace: tc.Host.Namespace,
+				},
+				&savedHost,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			if tc.ExpectPresent {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).NotTo(BeNil())
+			} else {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).To(Equal(""))
+			}
+		},
+		Entry("Remove BMH Pause Annotation, with valid CAPM3 Paused annotations", testCaseCheckPauseAnnotation{
+			Cluster: newCluster(clusterName),
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaWithValidCAPM3PausedAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: false,
+			ExpectError:   false,
+		}),
+		Entry("Do not Remove Annotation, with valid Paused annotations, Empty Key", testCaseCheckPauseAnnotation{
+			Cluster: newCluster(clusterName),
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaWithValidEmptyPausedAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: true,
+			ExpectError:   false,
+		}),
+	)
```

Comment:
you should add an entry without annotation, just to make sure nothing crashes if the annotation was removed unexpectedly.

Reply:
good point. I'll add it.